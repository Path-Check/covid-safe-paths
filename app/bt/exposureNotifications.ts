import dayjs from 'dayjs';

import { Possible, NoKnown, ExposureHistory } from '../ExposureHistoryContext';

type Posix = number;
type UUID = string;

export interface RawExposure {
  date: Posix;
  duration: number;
  id: UUID;
  totalRiskScore: number;
  transmissionRiskLevel: number;
}

export const toExposureHistory = (
  rawExposures: RawExposure[],
): ExposureHistory => {
  const possibleExposures = rawExposures.map(toPossible);
  const byDate = groupedByDate(possibleExposures, combinePossibles);

  const base = blankHistory();

  return base.map((datum: NoKnown) => {
    const date = datum.date;
    if (byDate[date]) {
      return byDate[date];
    } else {
      return datum;
    }
  });
};

const toPossible = (r: RawExposure): Possible => {
  const beginngingOfDay = (date: Posix) => dayjs(date).startOf('day');
  return {
    kind: 'Possible',
    date: beginngingOfDay(r.date).valueOf(),
    duration: r.duration,
    transmissionRiskLevel: r.transmissionRiskLevel,
    totalRiskScore: r.totalRiskScore,
  };
};

const combinePossibles = (a: Possible, b: Possible): Possible => {
  return {
    ...a,
    duration: a.duration + b.duration,
    totalRiskScore: Math.max(a.totalRiskScore, b.totalRiskScore),
    transmissionRiskLevel: Math.max(
      a.transmissionRiskLevel,
      b.transmissionRiskLevel,
    ),
  };
};

const groupedByDate = (
  exposures: Possible[],
  combine: (a: Possible, b: Possible) => Possible,
): Record<Posix, Possible> => {
  return exposures.reduce<Record<Posix, Possible>>((result, exposure) => {
    const date = exposure.date;
    if (result[date]) {
      result[date] = combine(result[date], exposure);
    } else {
      result[date] = exposure;
    }
    return result;
  }, {});
};

const blankHistory = (): NoKnown[] => {
  const now = Date.now();
  const daysAgo = [...Array(21)].map((_v, idx: number) => {
    return 20 - idx;
  });

  return daysAgo.map(
    (daysAgo: number): NoKnown => {
      const date = dayjs(now).subtract(daysAgo, 'day').startOf('day').valueOf();
      return {
        kind: 'NoKnown',
        date,
      };
    },
  );
};
