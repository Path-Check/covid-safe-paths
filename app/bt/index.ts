import { TracingStrategy } from '../tracingStrategy';
import * as BTNativeModule from './nativeModule';
import { PermissionsProvider } from './PermissionsContext';
import Home from './Home';
import AffectedUserFlow from './AffectedUserFlow';
import { useBTCopyContent, btAssets } from './content';

const btStrategy: TracingStrategy = {
  name: 'bt',
  exposureInfoSubscription: BTNativeModule.subscribeToExposureEvents,
  permissionsProvider: PermissionsProvider,
  homeScreenComponent: Home,
  affectedUserFlow: AffectedUserFlow,
  assets: btAssets,
  useCopy: useBTCopyContent,
};

export { BTNativeModule };
export default btStrategy;
