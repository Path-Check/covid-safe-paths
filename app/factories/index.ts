import { register } from 'fishery';
import tracingStrategy from './tracingStrategy';

export const factories = register({
  tracingStrategy,
});
