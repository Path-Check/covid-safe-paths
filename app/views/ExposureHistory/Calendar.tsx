import React from 'react';
import { Text, TouchableOpacity, View, StyleSheet } from 'react-native';
import dayjs from 'dayjs';

import { ExposureHistory, ExposureDatum } from '../../exposureHistory';
import { Typography } from '../../components/Typography';
import ExposureDatumIndicator from './ExposureDatumIndicator';

import { Colors, Spacing, Typography as TypographyStyles } from '../../styles';

interface CalendarProps {
  exposureHistory: ExposureHistory;
  onSelectDate: (exposureDatum: ExposureDatum) => void;
  selectedDatum: ExposureDatum | null;
}

const Calendar = ({
  exposureHistory,
  onSelectDate,
  selectedDatum,
}: CalendarProps): JSX.Element => {
  const lastMonth = dayjs().subtract(1, 'month');
  const title = `${lastMonth.format('MMMM')} | ${dayjs().format(
    'MMMM',
  )}`.toUpperCase();

  const week1 = exposureHistory.slice(0, 7);
  const week2 = exposureHistory.slice(7, 14);
  const week3 = exposureHistory.slice(14, 21);

  interface CalendarRowProps {
    week: ExposureHistory;
  }

  const CalendarRow = ({ week }: CalendarRowProps) => {
    return (
      <View style={styles.calendarRow}>
        {week.map((datum: ExposureDatum) => {
          return (
            <TouchableOpacity
              key={`calendar-day-${datum.date}`}
              onPress={() => onSelectDate(datum)}>
              <ExposureDatumIndicator
                isSelected={
                  selectedDatum ? datum.date === selectedDatum.date : false
                }
                exposureDatum={datum}
              />
            </TouchableOpacity>
          );
        })}
      </View>
    );
  };

  const DayLabels = () => {
    const labels = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
    return (
      <View style={styles.calendarRow}>
        {labels.map((label: string, idx: number) => {
          return (
            <View style={styles.labelStyle} key={`calendar-label-${idx}`}>
              <Text>{label}</Text>
            </View>
          );
        })}
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Typography style={styles.monthText}>{title}</Typography>
      </View>
      <View style={styles.calendarContainer}>
        <DayLabels />
        <CalendarRow week={week1} />
        <CalendarRow week={week2} />
        <CalendarRow week={week3} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  monthText: {
    ...TypographyStyles.label,
    color: Colors.secondaryHeaderText,
  },
  calendarContainer: { flex: 1, paddingVertical: Spacing.small },
  calendarRow: {
    flex: 1,
    paddingVertical: Spacing.xSmall,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  labelStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 36,
  },
});

export default Calendar;
