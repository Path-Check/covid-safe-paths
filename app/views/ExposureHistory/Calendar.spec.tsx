import React from 'react';
import { cleanup, render } from '@testing-library/react-native';

import { factories } from '../../factories';

import Calendar from './Calendar';
import { toExposureHistory } from '../../bt/exposureNotifications';

afterEach(cleanup);

describe('Calendar', () => {
  it('renders', () => {
    const exposureHistory = buildExposureHistory();
    const onSelectDate = () => {};
    const selectedDatum = exposureHistory[0];

    const { asJSON } = render(
      <Calendar
        exposureHistory={exposureHistory}
        onSelectDate={onSelectDate}
        selectedDatum={selectedDatum}
      />,
    );

    expect(asJSON()).toMatchSnapshot();
  });
});

const buildExposureHistory = () => {
  const rawExposure = factories.rawExposure.build();
  return toExposureHistory([rawExposure], {
    startDate: Date.now(),
    totalDays: 21,
  });
};
