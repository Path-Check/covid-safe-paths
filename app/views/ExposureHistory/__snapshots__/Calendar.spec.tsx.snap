// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontFamily": "IBMPlexSans",
              "fontSize": 17,
              "lineHeight": 28,
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "color": "#4754C5",
              "fontFamily": "IBMPlexSans-Bold",
              "fontSize": 15,
              "fontWeight": "500",
              "lineHeight": 24,
            },
          ]
        }
      >
        JUNE | JULY
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e5e7fa",
            "borderColor": "#e5e7fa",
            "borderRadius": 500,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 4,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 17,
                "lineHeight": 28,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#4754C5",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 15,
                "fontWeight": "500",
                "lineHeight": 24,
              },
            ]
          }
        >
          LEGEND
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            S
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            M
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            W
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            F
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 15,
                "lineHeight": 24,
              }
            }
          >
            S
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 8,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#4e4e4e",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              14
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              15
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              16
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              17
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              18
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              19
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              20
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 8,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              21
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              22
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              23
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              24
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              25
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              26
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              27
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 8,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              28
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              29
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4051db",
                "borderColor": "#4051db",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              30
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "700",
                  "lineHeight": 24,
                }
              }
            >
              2
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2e2e2e",
                  "borderRadius": 6,
                  "bottom": -4,
                  "height": 6,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 6,
                }
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              3
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0)",
                "borderRadius": 500,
                "borderWidth": 4,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1f2c9b",
                  "fontFamily": "IBMPlexMono",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "lineHeight": 24,
                }
              }
            >
              4
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.7)",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 16,
              "padding": 16,
              "paddingHorizontal": 24,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 100,
                    "height": 30,
                    "justifyContent": "center",
                    "marginRight": 16,
                    "marginVertical": 16,
                    "width": 30,
                  },
                  Object {
                    "backgroundColor": "#ffcc00",
                    "borderColor": "#ffcc00",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontWeight": "500",
                  }
                }
              >
                1
              </Text>
              <View
                style={null}
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "lineHeight": 24,
                  },
                ]
              }
            >
              Exposure very likely
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 100,
                    "height": 30,
                    "justifyContent": "center",
                    "marginRight": 16,
                    "marginVertical": 16,
                    "width": 30,
                  },
                  Object {
                    "backgroundColor": "#4051db",
                    "borderColor": "#4051db",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontWeight": "500",
                  }
                }
              >
                2
              </Text>
              <View
                style={null}
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "lineHeight": 24,
                  },
                ]
              }
            >
              Exposure possible
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 100,
                    "height": 30,
                    "justifyContent": "center",
                    "marginRight": 16,
                    "marginVertical": 16,
                    "width": 30,
                  },
                  null,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1f2c9b",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontWeight": "500",
                  }
                }
              >
                3
              </Text>
              <View
                style={null}
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "lineHeight": 24,
                  },
                ]
              }
            >
              No exposure detected
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 100,
                    "height": 30,
                    "justifyContent": "center",
                    "marginRight": 16,
                    "marginVertical": 16,
                    "width": 30,
                  },
                  null,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1f2c9b",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "lineHeight": 24,
                  }
                }
              >
                4
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4051db",
                    "borderRadius": 2,
                    "bottom": -2,
                    "height": 4,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 4,
                  }
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#2e2e2e",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "lineHeight": 24,
                  },
                ]
              }
            >
              Today
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
